rust/processor/src/db/common/models/default_models/parquet_write_set_changes.rs

```diff
@@ -58,98 +58,117 @@ impl Default for WriteSetChange {
-                        hex::encode(inner.state_key_hash.as_slice()).as_str(),
-                    ),
-                    block_height,
-                    change_type,
-                    resource_address: standardize_address(&inner.address.to_string()),
-                    write_set_change_index,
-                    block_timestamp,
-                },
-                WriteSetChangeDetail::Resource(MoveResource::from_delete_resource(
+            WriteSetChangeEnum::WriteModule(inner) => {
+                return Ok(Some((
+                    Self {
+                        txn_version,
+                        state_key_hash: standardize_address(
+                            hex::encode(inner.state_key_hash.as_slice()).as_str(),
+                        ),
+                        block_height,
+                        change_type,
+                        resource_address: standardize_address(&inner.address.to_string()),
+                        write_set_change_index,
+                        block_timestamp,
+                    },
+                    WriteSetChangeDetail::Module(MoveModule::from_write_module(
+                        inner,
+                        write_set_change_index,
+                        txn_version,
+                        block_height,
+                    )),
+                )));
+            },
+            WriteSetChangeEnum::DeleteModule(inner) => {
+                return Ok(Some((
+                    Self {
+                        txn_version,
+                        state_key_hash: standardize_address(
+                            hex::encode(inner.state_key_hash.as_slice()).as_str(),
+                        ),
+                        block_height,
+                        change_type,
+                        resource_address: standardize_address(&inner.address.to_string()),
+                        write_set_change_index,
+                        block_timestamp,
+                    },
+                    WriteSetChangeDetail::Module(MoveModule::from_delete_module(
+                        inner,
+                        write_set_change_index,
+                        txn_version,
+                        block_height,
+                    )),
+                )));
+            },
+            WriteSetChangeEnum::WriteResource(inner) => {
+                let resource_option = MoveResource::from_write_resource(
                     inner,
                     write_set_change_index,
                     txn_version,
                     block_height,
                     block_timestamp,
-                )),
-            ),
+                );
+
+                if let Some(resource) = resource_option.unwrap() {
+                    return Ok(Some((
+                        Self {
+                            txn_version,
+                            state_key_hash: standardize_address(
+                                hex::encode(inner.state_key_hash.as_slice()).as_str(),
```

Comment:
We should make a version of `standardize_address` that takes in a slice, so we don't have to do this hex::encode first everywhere.